src/OrbitDB.js

```diff
@@ -337,6 +375,19 @@ class OrbitDB {
     options = Object.assign({ localOnly: false, create: false }, options)
     logger.debug(`Open database '${address}'`)
 
+    if (options.directory) {
+      if (this.caches[options.directory]) {
+        options.cache = this.caches[options.directory]
+      } else {
+        this.caches[options.directory] = await this._createCache(options.directory)
+        options.cache = this.caches[options.directory]
+      }
+    } else {
+      options.cache = this.cache
+    }
+    // "Wake up" the caches if they need it
+    if (options.cache) await options.cache.open()
```

Comment:
Could this part be pulled out to its own function since we use exactly the same code above too?